File: e-shop-admin/ecart/src/main/java/com/ecart/searvice/ProductService.java
/**
 * 
 */
package com.ecart.searvice;

import java.util.List;

import com.ecart.domain.Product;
import com.ecart.model.CommonCustomResponce;

/**
 * @author Tharindu   Feb 23, 2020
 *
 */
public interface ProductService {

	/**
	 * @param product
	 * @return
	 */
	CommonCustomResponce createUpdate(Product product);

	/**
	 * @return
	 */
	List<Product> getAll();

	/**
	 * @param id
	 * @return
	 */
	Product getById(Long id);

	/**
	 * @param id
	 * @return
	 */
	CommonCustomResponce delete(Long id);

}

File: e-shop-admin/core-ui/src/app/views/product-management/product/service/product.service.ts
import { Injectable, SkipSelf, Optional } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { mainURL } from '../../../Common/common-url-main';
import { CommonCustomResponce } from '../../../common/common-custom-responce';
import { Product } from '../model/product';

@Injectable({
    providedIn: 'root'
  })
export class ProductService{
    url : string = mainURL+"product";

    constructor(private _http: HttpClient){} //@SkipSelf()  


    save(mainModel : Product){
        return this._http.post<CommonCustomResponce>(this.url,mainModel);
    }

    delete(id: any) {
        return this._http.delete<CommonCustomResponce>(this.url+"/"+id);
      }
     
    getAll() {
        return this._http.get<Product[]>(this.url);
    }

    getById(id: any) {
        return this._http.get<Product>(this.url+"/"+id);
    }
      

}